tests/ballerina-spec-conformance-tests/src/test/java/org/ballerinalang/test/TestRunnerUtils.java

```diff
@@ -196,6 +199,36 @@ private static String validateKindOfTest(String kind) {
         }
     }
 
+    public static void validateLabels(String labels, Set<String> predefinedLabels) {
+        List<String> labelsList = new ArrayList<>();
+        StringJoiner duplicateLabels = new StringJoiner(", ");
+        StringJoiner unknownLabels = new StringJoiner(", ");
+
+        for (String label : labels.split(",")) {
+            String trimmedLabel = label.trim();
+            if (labelsList.contains(trimmedLabel)) {
+                duplicateLabels.add(trimmedLabel);
+                continue;
+            }
+            if (!predefinedLabels.contains(trimmedLabel)) {
+                unknownLabels.add(trimmedLabel);
+            }
+            labelsList.add(trimmedLabel);
+        }
+
+        boolean hasUnknownLabels = unknownLabels.length() > 0;
+        boolean hasDuplicateLabels = duplicateLabels.length() > 0;
+
+        String diagnosticMsg = hasUnknownLabels ? (hasDuplicateLabels ?
```

Comment:
Shall we propagate the line number here and log it.

Reply:
Do you mean the line number of the test case or the line number of the label? We don't seem to keep information on the line numbers of the labels ATM though. Will need to add support for this if that's the requirement here.

Reply:
Line number in the test case. Specifically .balt file.
Isn't `absLineNum` argument TestRunner.test() represent that?

Reply:
Data provider provides the `absLineNum`. The line number of the last row of the labels will be  `absLineNum - 1` right?

Reply:
Yeah. But anyway we can see which test has failed right?. Then, we can just check and correct the label there. If we can't pinpoint the row I don't think it makes any difference, even if log the error to the last line.

Reply:
Pin-pointing the raw is not needed. Just the line number will be useful as I see it. It's matter of checking the label string in the test. 
Does the failed test pin-point to the specific line on the original balt. No right? 
I think logging the `absLineNum` would be enough for the moment and it will be really useful. Theat's what I felt trying out your changes against .balt files

Reply:
As discussed offline, added the line number in 7f9e8ab